streaming/storage/download.py

```diff
@@ -474,18 +475,25 @@ def download_file(remote: Optional[str], local: str, timeout: float):
         download_from_local(remote, local)
 
 
-def wait_for_download(local: str, timeout: float = 60) -> None:
-    """Wait for a download by another thread/process to complete.
+def wait_for_file_to_exist(filename: str, poll_interval: float, timeout: float,
```

Comment:
Can we keep this method in util directory? Looks like this method does not fit well here with other cloud provider download methods.

Reply:
should this maybe go in `retrying.py`?

Reply:
This move portends future PRs that encapsulate the complexity of downloading with size check/hash validation and other "smart" functionality within the storage APIs, simplifying StreamingDataset proper -- wdyt in this context?

Reply:
Would prefer to punt on messing with this to after future PRs are in and have fleshed things out?